File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Nodejs/core/handler/CreateClassInstanceHandler.js
const AbstractHandler = require("./AbstractHandler");

class CreateClassInstanceHandler extends AbstractHandler {
  requiredParametersCount = 1;

  constructor() {
    super();
  }

  process(command) {
    try {
      if (command.payload.length < this.requiredParametersCount) {
        throw new Error("Create Class Instance parameters mismatch");
      }
      let clazz = command.payload[0];
      let constructorArguments = command.payload.slice(1);
      let instance = new clazz(...constructorArguments);
      if (typeof instance === "undefined") {
        let methods = Object.getOwnPropertyNames(type).filter(function (
          property
        ) {
          return typeof type[property] === "function";
        });
        let message = `Method 'constructor' not found in class. Available methods:\n`;
        methods.forEach((methodIter) => {
          message += `${methodIter}\n`;
        });
        throw new Error(message);
      } else {
        return instance;
      }
    } catch (error) {
      throw this.process_stack_trace(error, this.constructor.name);
    }
  }
}

module.exports = new CreateClassInstanceHandler();

File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Ruby/Windows/X64/core/handler/create_class_instance_handler.rb
require_relative 'abstract_command_handler'

class CreateClassInstanceHandler < AbstractCommandHandler
  def initialize
    @required_parameters_count = 1
  end

  def process(command)
    return create_class_instance(command)
  end

  def create_class_instance(command)
    begin
      if command.payload.length < @required_parameters_count
        raise ArgumentError.new "Class instance parameters mismatch"
      end
      if command.payload.length > 1
        constructor_arguments = command.payload[1..]
        class_instance = command.payload[0].send('new', *constructor_arguments)
      else
        class_instance = command.payload[0].send('new')
      end
      return class_instance
    rescue NoMethodError
      methods = command.payload[0].methods
      message = "Method 'new' not found in class #{command.payload[0].name}. Available methods:\n"
      methods.each { |method_iter| message += "#{method_iter}\n" }
      raise Exception, message
    rescue Exception => e
      return e
    end
  end

end